packages/debug/src/browser/debug-session.tsx

```diff
@@ -93,7 +93,17 @@ export class DebugSession implements CompositeTreeElement {
                 }
             }),
             this.on('stopped', async ({ body }) => {
+                // Update thread list
                 await this.updateThreads(body);
+
+                if (body.allThreadsStopped) {
+                    // Mark all threads as having dirty frames that need to be re-fetched on click.
+                    for (const thread of this.threads) {
```

Comment:
Generally, I think the fact that we're actively telling threads that their frames are dirty is a bit of a code smell: shouldn't threads be aware when their own state changes (from `running` to `stopped`) and decide themselves what to do?

Reply:
Your comment can mean two things to me:

1. Stopped event would be handled at least in part in the thread implementation?
2. Do what you mentioned: clear frames and fetch new ones when empty and accessed?

Reply:
I mean both.

Reply:
Thank you, I moved most logic to the thread class :)